Use browser language

imenattatra opened the pull request:
## What type of PR is this? 

- [x] Feature


## Description
Closes https://github.com/metr-systems/backlog/issues/3703

This pull request includes changes to the localization files for the English language and an update to the language selection logic in the initialisation of the i18n instance. 

We Updated the language selection logic to dynamically choose between English and German based on the user's browser language settings.


## How is this tested?

- [x] Manually ( [staging](https://dashboard.staging.metr.systems/staging/login?next=https://dashboard.staging.metr.systems/staging/) ) 
- [x] Also checked that unit tests are running 

## UI

You can check it on [staging](https://dashboard.staging.metr.systems/staging/login?next=https://dashboard.staging.metr.systems/staging/). If you want to test it : 
- go to chrome and set to top the language you prefer (en/de)
- refresh redash page , it should show up in the language you select (please notice that the login page is translated in backend and the dashboards is translated in the frontend)

I would be very thankful , if you check it also in your machines especially if you are using other browsers like edge.

![](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExZTJubWViaDhvYnB1ZGJzMm03ZnQxeWY5ZXV3eWNvdWFvY3VpcHBydyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/dAWbBIUwTU0E5dxJEf/giphy.gif)

imenattatra committed:
Adding values to english files

Even if values are same as keys and in i18n configuration i am specifying to fallback to the keys rather than empty values.
Since the german values exists it priritize german values rather than the keys.
I did not find yet how to change this priority ..also to be more sure about what happens I filled the values.

imenattatra committed:
set i18n  language according to the navigator language
i experimented with Language Detector before but since we need a much simpler behavior this one is enough

imenattatra committed:
Use the browser language also for the backend

helenalebreton commented:
@imenattatra I am currently testing in english with Edge and Chrome.
The login page is in English, inside  redash is in english for both.
We have a few translation escapees but all of the same type so maybe there is a pattern somewhere in the code

https://dashboard.staging.metr.systems/staging/users?order=-created_at&page=1&page_size=20
![image](https://github.com/user-attachments/assets/6bbc9706-b79d-43b4-98c8-e3a05c742783)

https://dashboard.staging.metr.systems/staging/queries/15?p_date_range=d_this_year

interstingly the refresh schedule at query and dashboard level do not seem impacted
![image](https://github.com/user-attachments/assets/4b371126-c871-421f-9160-054611d28e88)


This happens in both browsers.

I will look at switching to German a bit later

imenattatra committed:
[FIX] update moment with the correct language

imenattatra committed:
Add missing translation of the setup part
The setup is when we setup a new organization, should appear when you run
the app the first time

imenattatra commented:
> @imenattatra I am currently testing in english with Edge and Chrome. The login page is in English, inside redash is in english for both. We have a few translation escapees but all of the same type so maybe there is a pattern somewhere in the code
> 
> https://dashboard.staging.metr.systems/staging/users?order=-created_at&page=1&page_size=20 ![image](https://private-user-images.githubusercontent.com/77674724/406054048-6bbc9706-b79d-43b4-98c8-e3a05c742783.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc3MTg2OTQsIm5iZiI6MTczNzcxODM5NCwicGF0aCI6Ii83NzY3NDcyNC80MDYwNTQwNDgtNmJiYzk3MDYtYjc5ZC00M2I0LTk4YzgtZTNhMDVjNzQyNzgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI0VDExMzMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1OTVjODFlNjdkM2ZkM2EwZmU4YzFiNzIwM2I5OTA1MDdjYTExOWMzMWVkOWQ5YzM4YzNiZDM3ZjA3ZmJmZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9PiZGgVZDeJq4_Ojn0JLfQ98GyT7yBQrVnCrvO6Ihrg)
> 
> https://dashboard.staging.metr.systems/staging/queries/15?p_date_range=d_this_year
> 
> interstingly the refresh schedule at query and dashboard level do not seem impacted ![image](https://private-user-images.githubusercontent.com/77674724/406054785-4b371126-c871-421f-9160-054611d28e88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc3MTg2OTQsIm5iZiI6MTczNzcxODM5NCwicGF0aCI6Ii83NzY3NDcyNC80MDYwNTQ3ODUtNGIzNzExMjYtYzg3MS00MjFmLTkxNjAtMDU0NjExZDI4ZTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI0VDExMzMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkZGZkOWJmNjMyMDAwZGJiNjUyMTBkYzhlZmZhOWI1NzM4MmYzNTUxYzAzZTA3MWU4ODBlNGI4NzNhNmIyZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fLyYeH032xFdW2C8J17sdp4IxVOdBrD3syNLrlQKHhQ)
> 
> This happens in both browsers.
> 
> I will look at switching to German a bit later

It is fixed now :D thank you, I deployed the updates to staging

helenalebreton commented:
@imenattatra German is looking good in all browsers (Edge Firefox and Chrome). English is fixed at least in Edge and Chrome.
I am pretty sure it will be as well in Firefox.
